File: leetcode-python/weeklycontest161/problem2.py
from typing import List


class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        odd_flag = [0 if e % 2 == 0 else 1 for e in nums]
        n = len(nums)
        t = 0
        for j in range(n):
            t += odd_flag[j]
            if t == k:
                break
        i, j = 0, t
        res = 0
        while True:
            # 实现太混乱
            pass

        return res


s = Solution()
nums = [2, 2, 2, 1, 2, 2, 1, 2, 2, 2]
print(s.numberOfSubarrays(nums, 2))

File: leetcode-cpp/weeklycontest/218/problem2.cpp
#include <algorithm>
#include <deque>
#include <iostream>
#include <regex>
#include <stack>
#include <unordered_map>
#include <vector>

using namespace std;

bool compare(const pair<int, int> &e1, const pair<int, int> &e2) {
  return e1.second > e2.second;
}
class Solution {
public:
  int maxOperations(vector<int> &nums, int k) {
    unordered_map<int, int> num_cnt;
    for (auto num : nums) {
      num_cnt[num]++;
    }
    vector<pair<int, int>> numcnt;
    for (auto kv : num_cnt) {
      numcnt.push_back(kv);
    }
    sort(numcnt.begin(), numcnt.end(), compare);

    int res = 0;
    for (auto kv : numcnt) {
      if (num_cnt[kv.first] > 0 && num_cnt[k - kv.first] > 0) {
        if (kv.first * 2 == k) {
          res += kv.second / 2;
          num_cnt[kv.first] = 0;
        } else {
          res += num_cnt[k - kv.first];
          num_cnt[kv.first] = num_cnt[kv.first] - num_cnt[k - kv.first];
          num_cnt[k - kv.first] = 0;
        }
      }
    }

    return res;
  }
};

int main() {
  Solution s;
  vector<int> nums = {2, 2, 2, 3, 1, 1, 4, 1};
  int k = 4;
  cout << s.maxOperations(nums, k) << endl;
}